File: comparison_tests/test_quick_scenarios.py
#!/usr/bin/env python3
"""
Quick test scenarios for MOCAT-MC Python implementation
Runs key tests with reduced time steps for faster execution
"""

import numpy as np
import sys
import os
import time

# Add parent directory to path
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from mocat_mc import MOCATMC


def run_quick_test(test_name, seed, ic_file='2020.mat'):
    """Run a quick test scenario"""
    mocat = MOCATMC()
    
    print(f"\n{'='*50}")
    print(f"Test: {test_name}")
    print(f"Seed: {seed}")
    print(f"{'='*50}")
    
    try:
        start_time = time.time()
        
        # Setup base configuration
        cfg_mc = mocat.setup_mc_config(seed, ic_file)
        
        # Apply test-specific settings
        if test_name == "Basic Propagation":
            cfg_mc['n_time'] = 5  # Very short
            cfg_mc['skipCollisions'] = 1
            cfg_mc['P_frag'] = 0
            cfg_mc['launch_model'] = 'no_launch'
            
        elif test_name == "Collision Test":
            cfg_mc['n_time'] = 10
            cfg_mc['dt_days'] = 5
            cfg_mc['skipCollisions'] = 0
            cfg_mc['CUBE_RES'] = 50
            cfg_mc['launch_model'] = 'no_launch'
            
        elif test_name == "Launch Test":
            cfg_mc['n_time'] = 20
            cfg_mc['launch_model'] = 'matsat'
            cfg_mc['launchRepeatYrs'] = [2018, 2020]
            cfg_mc['skipCollisions'] = 1
            
        elif test_name == "Fragmentation Test":
            cfg_mc['n_time'] = 10
            cfg_mc['P_frag'] = 1e-4  # Higher probability for testing
            cfg_mc['max_frag'] = 50
            cfg_mc['skipCollisions'] = 1
            
        elif test_name == "Full Integration":
            cfg_mc['n_time'] = 15
            # All features enabled with defaults
            
        # Get initial stats
        initial_pop = cfg_mc['mat_sats'].shape[0]
        
        # Run simulation
        nS, nD, nN, nB, mat_sats = mocat.main_mc(cfg_mc, seed)
        
        # Calculate results
        elapsed_time = time.time() - start_time
        total_objects = nS + nD + nN + nB
        
        # Print results
        print(f"Initial Population: {initial_pop}")
        print(f"Final Population: {total_objects}")
        print(f"Change: {total_objects - initial_pop:+d}")
        print(f"S={nS}, D={nD}, N={nN}, B={nB}")
        print(f"Time: {elapsed_time:.2f}s")
        
        return {
            'test': test_name,
            'success': True,
            'initial': initial_pop,
            'final': total_objects,
            'nS': nS, 'nD': nD, 'nN': nN, 'nB': nB,
            'time': elapsed_time
        }
        
    except Exception as e:
        print(f"ERROR: {str(e)}")
        return {
            'test': test_name,
            'success': False,
            'error': str(e)
        }


def main():
    """Run quick test suite"""
    print("MOCAT-MC Python Quick Test Suite")
    print("="*50)
    
    tests = [
        ("Basic Propagation", 42),
        ("Collision Test", 123),
        ("Launch Test", 200),
        ("Fragmentation Test", 999),
        ("Full Integration", 1)
    ]
    
    results = []
    for test_name, seed in tests:
        result = run_quick_test(test_name, seed)
        results.append(result)
    
    # Print summary
    print("\n" + "="*50)
    print("SUMMARY")
    print("="*50)
    
    print(f"\n{'Test':<20} {'Status':<10} {'Initial':<8} {'Final':<8} {'Time':<8}")
    print("-"*60)
    
    for r in results:
        if r['success']:
            print(f"{r['test']:<20} {'PASS':<10} {r['initial']:<8} {r['final']:<8} {r['time']:<8.2f}s")
        else:
            print(f"{r['test']:<20} {'FAIL':<10} Error: {r.get('error', '')}")
    
    # Save results
    import json
    with open('quick_test_results.json', 'w') as f:
        json.dump(results, f, indent=2)
    print("\nResults saved to quick_test_results.json")


if __name__ == "__main__":
    main()

File: MATLAB_implementation/test_quick_scenarios.m
%% Quick Test Scenarios for MOCAT-MC MATLAB
% Reduced test suite for faster execution

function test_quick_scenarios()
    % Add paths
    addpath(genpath('../supporting_data/')); 
    addpath(genpath('../supporting_functions'));
    
    fprintf('MOCAT-MC MATLAB Quick Test Suite\n');
    fprintf('%s\n', repmat('=', 1, 50));
    
    % Quick tests with reduced time steps
    tests = {
        'Basic Propagation',  42;
        'Collision Test',     123;
        'Launch Test',        200;
        'Fragmentation Test', 999;
        'Full Integration',   1;
    };
    
    results = [];
    
    for i = 1:size(tests, 1)
        test_name = tests{i, 1};
        seed = tests{i, 2};
        
        fprintf('\n%s\n', repmat('=', 1, 50));
        fprintf('Test: %s\n', test_name);
        fprintf('Seed: %d\n', seed);
        fprintf('%s\n', repmat('=', 1, 50));
        
        try
            tic;
            
            % Setup base configuration
            cfgMC = setup_MCconfig(seed, '2020.mat');
            
            % Apply test-specific settings (reduced for speed)
            switch test_name
                case 'Basic Propagation'
                    cfgMC.n_time = 5;  % Very short
                    cfgMC.skipCollisions = 1;
                    cfgMC.P_frag = 0;
                    cfgMC.launch_model = 'no_launch';
                    
                case 'Collision Test'
                    cfgMC.n_time = 10;
                    cfgMC.dt_days = 5;
                    cfgMC.skipCollisions = 0;
                    cfgMC.CUBE_RES = 50;
                    cfgMC.launch_model = 'no_launch';
                    
                case 'Launch Test'
                    cfgMC.n_time = 20;
                    cfgMC.launch_model = 'matsat';
                    cfgMC.launchRepeatYrs = [2018, 2020];
                    cfgMC.skipCollisions = 1;
                    
                case 'Fragmentation Test'
                    cfgMC.n_time = 10;
                    cfgMC.P_frag = 1e-4;  % Higher for testing
                    cfgMC.max_frag = 50;
                    cfgMC.skipCollisions = 1;
                    
                case 'Full Integration'
                    cfgMC.n_time = 15;
                    % All defaults
            end
            
            % Get initial stats
            initial_pop = size(cfgMC.mat_sats, 1);
            
            % Run simulation
            [nS, nD, nN, nB, mat_sats] = main_mc(cfgMC, seed);
            
            % Calculate results
            elapsed_time = toc;
            total_objects = nS + nD + nN + nB;
            
            % Store result
            result.test = test_name;
            result.success = true;
            result.initial = initial_pop;
            result.final = total_objects;
            result.nS = nS;
            result.nD = nD;
            result.nN = nN;
            result.nB = nB;
            result.time = elapsed_time;
            
            fprintf('Initial Population: %d\n', initial_pop);
            fprintf('Final Population: %d\n', total_objects);
            fprintf('Change: %+d\n', total_objects - initial_pop);
            fprintf('S=%d, D=%d, N=%d, B=%d\n', nS, nD, nN, nB);
            fprintf('Time: %.2fs\n', elapsed_time);
            
        catch ME
            fprintf('ERROR: %s\n', ME.message);
            result.test = test_name;
            result.success = false;
            result.error = ME.message;
        end
        
        results = [results, result];
    end
    
    % Print summary
    fprintf('\n%s\n', repmat('=', 1, 50));
    fprintf('SUMMARY\n');
    fprintf('%s\n', repmat('=', 1, 50));
    
    fprintf('\n%-20s %-10s %-8s %-8s %-8s\n', 'Test', 'Status', 'Initial', 'Final', 'Time');
    fprintf('%s\n', repmat('-', 1, 60));
    
    for i = 1:length(results)
        r = results(i);
        if r.success
            fprintf('%-20s %-10s %-8d %-8d %-8.2fs\n', ...
                r.test, 'PASS', r.initial, r.final, r.time);
        else
            fprintf('%-20s %-10s Error: %s\n', r.test, 'FAIL', r.error);
        end
    end
    
    % Save results
    save('quick_test_results.mat', 'results');
    fprintf('\nResults saved to quick_test_results.mat\n');
end